truffle/src/main/java/org/jruby/truffle/nodes/rubinius/StringPrimitiveNodes.java

```diff
@@ -125,16 +136,45 @@ public CharacterPrintablePrimitiveNode(RubyContext context, SourceSection source
         @TruffleBoundary
         @Specialization
         public boolean isCharacterPrintable(DynamicObject character) {
-            final ByteList bytes = StringOperations.getByteList(character);
-            final Encoding encoding = bytes.getEncoding();
+            final Rope rope = rope(character);
+            final Encoding encoding = rope.getEncoding();
 
-            final int codepoint = encoding.mbcToCode(bytes.unsafeBytes(), bytes.begin(), bytes.begin() + bytes.realSize());
+            final int codepoint = encoding.mbcToCode(rope.getBytes(), 0, rope.byteLength());
 
             return encoding.isPrint(codepoint);
         }
 
     }
 
+    @RubiniusPrimitive(name = "string_append")
+    public static abstract class StringAppendPrimitiveNode extends RubiniusPrimitiveNode {
```

Comment:
`String#concat` needs to deal with `Fixnum` values as well. Otherwise, they're essentially the same.